xdsl/dialects/experimental/stencil.py

```diff
@@ -61,6 +62,31 @@ def verify(self):
                 f"Number of field dimensions must be greater than zero, got {self.get_num_dims()}."
             )
 
+    @staticmethod
+    def parse_parameters(parser: Parser) -> list[Attribute]:
```

Comment:
Just as a note, this parsing will not work if you port it to MLIR.
This is because MLIR does not let you parse a shape in custom parsers. However, if we ever want to port stencil to MLIR, we could probably try to add it in their parser as well.


In any way, I would suggest to make `parser._parse_shape` public instead of using `parse_memref_attrs`, which I'll remove soon from the public API, especially with the hack that you are checking that no layout or memory_space is set ;)

Reply:
Big thumbs up from me for the latter suggestion!

Reply:
Sure, I went for the easy, self-contained path!
I would even use parse_ranked_shape 🙂
Would you approve making this/those public in this PR?

Reply:
Yes!